test/agentchat/test_cache.py

```diff
@@ -0,0 +1,109 @@
+import os
+import sys
+import time
+
+import pytest
+import autogen
+from autogen.agentchat import AssistantAgent, UserProxyAgent
+sys.path.append(os.path.join(os.path.dirname(__file__), ".."))
+from conftest import skip_openai, skip_redis  # noqa: E402
+
+try:
+    from openai import OpenAI
+except ImportError:
+    skip = True
+else:
+    skip = False or skip_openai
+
+try:
+    import redis
+except ImportError:
+    skip_redis = True
+else:
+    skip_redis = False or skip_redis
+
+@pytest.mark.skipif(skip, reason="openai not installed OR requested to skip")
+def test_disk_cache(human_input_mode="NEVER", max_consecutive_auto_reply=5):
+    random_cache_seed = int.from_bytes(os.urandom(2), "big")
+    start_time = time.time()
+    cold_cache_messages = run_conversation(cache_seed=random_cache_seed, human_input_mode=human_input_mode, max_consecutive_auto_reply=max_consecutive_auto_reply)
+    end_time = time.time()
+    duration_with_cold_cache = end_time - start_time
+
+    start_time = time.time()
+    warm_cache_messages = run_conversation(cache_seed=random_cache_seed, human_input_mode=human_input_mode, max_consecutive_auto_reply=max_consecutive_auto_reply)
+    end_time = time.time()
+    duration_with_warm_cache = end_time - start_time
+    assert cold_cache_messages == warm_cache_messages
+    assert duration_with_warm_cache < duration_with_cold_cache
+
+@pytest.mark.skipif(skip_redis, reason="redis not installed OR requested to skip")
```

Comment:
in CI we should probably have a test runner that actually runs this test, against a local redis instance where `REDIS_URL` is set in the runner's env.

Reply:
added to contrib-openai.yml github actions workflow